### ISSUE TITLE
force_terminate_repair rest api command fails to terminate nodetool repair

### ISSUE THREAD

**Issue by ShlomiBalalis** [scylladb/scylla#8831]
*Installation details*
Scylla version (or git commit hash): 4.6.dev-0.20210531.b0c22f2e39 with build-id 6a2a6e52891ef3f81d9a844a9b84d45da27c6cbe
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI):ami-0c8cdedf851cde9cf (eu-north-1)

At `2021-06-05 01:00:09.468`, an AbortRepairMonkey nemesis ran on node 5 (13.51.69.232 | 10.0.0.197). The nemesis starts a standard nodetool repair, wait for the repair to start on the node, and afterward attempts to stops it with a rest api post:
`curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' http://127.0.0.1:10000/storage_service/force_terminate_repair`

At `2021-06-05T01:00:15`, the node has started the repair:
```
2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace keyspace1, repair id [id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], options {{ trace -> false}, { jobThreads -> 1}, { incremental -> false}, { parallelism -> parallel}, { primaryRange -> false}}2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - repair id [id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f]: started replaying hints before repair, source nodes: {10.0.1.55, 10.0.0.26, 10.0.0.197, 10.0.
0.112}, target nodes: {10.0.0.26, 10.0.1.55, 10.0.0.112, 10.0.0.197}
2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Aborted 1 repair job(s)
2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !WARNING | scylla:  [shard 0] repair - repair id [id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f]: failed to replay hints before repair (took 0s): seastar::sleep_aborted (Sleep is aborted), th
e repair will continue
2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Repair 1 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=0, keyspace=keyspace1, table={standard1}, range=(-9092081809196982
739, -9067818348342298713]
2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Repair 2 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=0, keyspace=keyspace1, table={standard1}, range=(-9067818348342298
713, -9063629502678422513]
2021-06-05T01:00:15+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Repair 3 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=0, keyspace=keyspace1, table={standard1}, range=(-9063629502678422
513, -9026340203877723606]
```

Soon after, SCT executed the force_terminate_repair command:
```
< t:2021-06-05 01:00:15,826 f:remote_base.py  l:520  c:RemoteCmdRunner      p:DEBUG > Running command "curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' http://127.0.0.1:10000/storage_service/force_terminate_repair"...
< t:2021-06-05 01:00:15,895 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' http://127.0.0.1:10000/storage_service/force_terminate_repair" finished with status 0
```

However, it appears that the command had no effect on the node, and while there was a message that states `Aborted 1 repair job(s)`, the repair did not stop even 15 minutes later, and it even ended successfully:
```
2021-06-05T01:16:35+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 12] repair - Repair 772 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=12, keyspace=keyspace1, table={standard1}, range=(9124283997254541273, 9155023026680603937]
2021-06-05T01:16:35+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Repair 771 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=0, keyspace=keyspace1, table={standard1}, range=(9095671708394733906, 9124283997254541273]
2021-06-05T01:16:36+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Repair 772 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=0, keyspace=keyspace1, table={standard1}, range=(9124283997254541273, 9155023026680603937]
2021-06-05T01:16:36+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 5] repair - Repair 766 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=5, keyspace=keyspace1, table={standard1}, range=(9010969948277304886, 9014501864962556156]
2021-06-05T01:16:36+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - Repair 773 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=0, keyspace=keyspace1, table={standard1}, range=(9155023026680603937, 9200027409839215776]
2021-06-05T01:16:37+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 12] repair - Repair 773 out of 773 ranges, id=[id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f], shard=12, keyspace=keyspace1, table={standard1}, range=(9155023026680603937, 9200027409839215776]
...
2021-06-05T01:16:47+00:00  longevity-200gb-48h-verify-limited--db-node-f420e06a-5 !INFO    | scylla:  [shard 0] repair - repair id [id=1, uuid=9d4f05e3-86bb-46f0-b755-05ec42a9426f] on shard 0 completed successfully, keyspace=keyspace1
```

This failure is consistent, and occurred 5 times in this run:
* 2021-06-05 01:00:09.468 to 2021-06-05 01:17:14.805 on node 5 (13.51.69.232 | 10.0.0.197)
* 2021-06-05 04:11:24.024 to 2021-06-05 04:30:04.346 on node 6 (13.48.1.50 | 10.0.0.112)
* 2021-06-05 05:01:09.450 to 2021-06-05 05:18:10.855 on node 4 (13.51.64.76 | 10.0.1.55)
* 2021-06-05 17:36:09.922 to 2021-06-05 17:47:56.707 on node 7 (13.48.105.245 | 10.0.2.194)
* 2021-06-05 22:25:54.416 to 2021-06-05 22:37:53.679 on node 7 (13.48.105.245 | 10.0.2.194)

Logs: 
```
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                                                                                           Log links for testrun with test id f420e06a-066e-4695-b85a-89fffe9fd50c                                                                                                            |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Date            | Log type    | Link                                                                                                                                                                                                                                                         |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 20210606_042600 | grafana     | https://cloudius-jenkins-test.s3.amazonaws.com/f420e06a-066e-4695-b85a-89fffe9fd50c/20210606_042600/grafana-screenshot-longevity-200gb-48h-scylla-per-server-metrics-nemesis-20210606_043009-longevity-200gb-48h-verify-limited--monitor-node-f420e06a-1.png |
| 20210606_043433 | db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/f420e06a-066e-4695-b85a-89fffe9fd50c/20210606_043433/db-cluster-f420e06a.zip                                                                                                                                  |
| 20210606_043433 | loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/f420e06a-066e-4695-b85a-89fffe9fd50c/20210606_043433/loader-set-f420e06a.zip                                                                                                                                  |
| 20210606_043433 | monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/f420e06a-066e-4695-b85a-89fffe9fd50c/20210606_043433/monitor-set-f420e06a.zip                                                                                                                                 |
| 20210606_043433 | sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/f420e06a-066e-4695-b85a-89fffe9fd50c/20210606_043433/sct-runner-f420e06a.zip                                                                                                                                  |
+-----------------+-------------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

Sadly, the pipeline of the job failed to create a snapshot of the monitor, so it cannot be linked here.

**Comment by roydahan** [comment]
@asias ?

**Comment by asias** [comment]
> Sadly, the pipeline of the job failed to create a snapshot of the monitor, so it cannot be linked here.

Don't worry. The monitor is not useful for debugging this anyway.

**Comment by asias** [comment]
It is a regression caused by 

```
commit 49f4a2f9689860b728c10284eb7ccc79743a2068
Author: Piotr Dulikowski <piodul@scylladb.com>
Date:   Wed Mar 10 12:27:52 2021 +0100

    repair: plug in waiting for hints to be sent before repair
    
    Now, the repair coordinator will wait before hints are sent between
    participating nodes before continuing with repair.
```

**Comment by fgelcer** [comment]
i'm seeing this issue again (or very similar) on version xxxxx

```
2021-06-18T04:46:40+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | scylla:  [shard 0] repair - starting user-requested repair for keyspace system_distributed_everywhere, repair id [id=1, uuid=ad327f2e-1bbd-4242-8b39-d8b6c7764fe2], options {{ trace -> false}, { jobThreads -> 1}, { incremental -> false}, { parallelism -> parallel}, { primaryRange -> false}}
2021-06-18T04:46:40+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | scylla:  [shard 0] repair - repair id [id=1, uuid=ad327f2e-1bbd-4242-8b39-d8b6c7764fe2]: started replaying hints before repair, source nodes: {10.0.3.56, 10.0.1.212, 10.0.3.249, 10.0.1.125}, target nodes: {10.0.1.125, 10.0.3.249, 10.0.1.212, 10.0.3.56}
2021-06-18T04:46:40+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | scylla:  [shard 0] repair - Aborted 1 repair job(s)
2021-06-18T04:46:40+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | scylla:  [shard 0] repair - repair id [id=1, uuid=ad327f2e-1bbd-4242-8b39-d8b6c7764fe2]: finished replaying hints (took 0s), continuing with repair
2021-06-18T04:46:40+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | scylla:  [shard 0] repair - Repair 1 out of 1025 ranges, id=[id=1, uuid=ad327f2e-1bbd-4242-8b39-d8b6c7764fe2], shard=0, keyspace=system_distributed_everywhere, table={cdc_generation_descriptions_v2}, range=(-inf, -9201768639713864622]
```
then, we run:
```
< t:2021-06-18 04:46:40,634 f:base.py         l:140  c:RemoteCmdRunner      p:DEBUG > Command "curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' http://127.0.0.1:
10000/storage_service/force_terminate_repair" finished with status 0
```

the repair continued to run for ~5 seconds, when apparently the node crashed (it restarted, but i cannot find any core, nor other message in the node's log, but this:)
```
2021-06-18T04:46:45+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | scylla:  [shard 0] repair - Repair 215 out of 748 ranges, id=[id=2, uuid=3be7ce7a-c5ee-4dd5-8f41-374ce232fcb7]
, shard=0, keyspace=system_auth, table={role_permissions, roles, role_attributes, role_members}, range=(-4518706384950728764, -4513940310492556629]
2021-06-18T05:25:13+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | systemd[1]: Mounted Huge Pages File System.
2021-06-18T05:25:13+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | systemd[1]: Mounted POSIX Message Queue File System.
2021-06-18T05:25:13+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | systemd[1]: Mounted Kernel Debug File System.
2021-06-18T05:25:13+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !INFO    | systemd[1]: Mounted Kernel Trace File System.
2021-06-18T05:25:13+00:00  longevity-200gb-48h-verify-limited--db-node-eadde21f-2 !NOTICE  | kernel: [    0.000000] Linux version 5.8.0-1035-aws (buildd@lgw01-amd64-022) (gcc (Ubuntu 9.3.0-17ubuntu1~20.0
4) 9.3.0, GNU ld (GNU Binutils for Ubuntu) 2.34) #37~20.04.1-Ubuntu SMP Tue Jun 1 09:54:15 UTC 2021 (Ubuntu 5.8.0-1035.37~20.04.1-aws 5.8.18)
```

SCT reported a core, but much later, but still didn't collect the core location, nor its content

full logs are here:
```
+--------------------------------------------------------------------------------------------------------------------------------------------+
|                                      Collected logs by test-id: eadde21f-ad93-476f-a546-842a4fea2708                                       |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
| Cluster set | Link                                                                                                                         |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
| db-cluster  | https://cloudius-jenkins-test.s3.amazonaws.com/eadde21f-ad93-476f-a546-842a4fea2708/20210620_043349/db-cluster-eadde21f.zip  |
| monitor-set | https://cloudius-jenkins-test.s3.amazonaws.com/eadde21f-ad93-476f-a546-842a4fea2708/20210620_043349/monitor-set-eadde21f.zip |
| loader-set  | https://cloudius-jenkins-test.s3.amazonaws.com/eadde21f-ad93-476f-a546-842a4fea2708/20210620_043349/loader-set-eadde21f.zip  |
| sct-runner  | https://cloudius-jenkins-test.s3.amazonaws.com/eadde21f-ad93-476f-a546-842a4fea2708/20210620_043349/sct-runner-eadde21f.zip  |
+-------------+------------------------------------------------------------------------------------------------------------------------------+
```

**Comment by asias** [comment]
@fgelcer it is a known issue. There is a PR to fix this issue. The PR is not fixed yet.

**Comment by piodul** [comment]
@fgelcer The PR which should fix this (https://github.com/scylladb/scylla/pull/8982) was recently merged to master. Could you check if the test passes now?

**Comment by fgelcer** [comment]
> @fgelcer The PR which should fix this (#8982) was recently merged to master. Could you check if the test passes now?

let's see this week's master result, although i see it did not pass, but someone from QA team shall report the analysis of the job this week, and i will add the comments to here... AFAIK this issue wasn't being reproduced on every run (or at least, it wasn't reported on every run), so we will wait couple more runs to confirm the issue was fixed...

**Comment by fgelcer** [comment]
> @fgelcer The PR which should fix this (#8982) was recently merged to master. Could you check if the test passes now?

@piodul , checked last jobs on master, and all of them that had the `AbortRepair` nemesis, all passed...
the specific job where i saw it reproducing is not passing for months now (`longevity-200gb-48h-test`)
so, we could say that the fix seems good..

**Comment by piodul** [comment]
> > @fgelcer The PR which should fix this (#8982) was recently merged to master. Could you check if the test passes now?
> 
> @piodul , checked last jobs on master, and all of them that had the `AbortRepair` nemesis, all passed...
> the specific job where i saw it reproducing is not passing for months now (`longevity-200gb-48h-test`)
> so, we could say that the fix seems good..

@fgelcer Thanks. In that case, I'm closing this issue.

**Comment by fgelcer** [comment]
> > > @fgelcer The PR which should fix this (#8982) was recently merged to master. Could you check if the test passes now?
> > 
> > 
> > @piodul , checked last jobs on master, and all of them that had the `AbortRepair` nemesis, all passed...
> > the specific job where i saw it reproducing is not passing for months now (`longevity-200gb-48h-test`)
> > so, we could say that the fix seems good..
> 
> @fgelcer Thanks. In that case, I'm closing this issue.

if seen again, a new issue should be reported.
thank you @piodul
